Sources/Sentry/SentrySessionReplay.m

```diff
@@ -98,8 +99,13 @@ - (void)start:(UIView *)rootView fullSession:(BOOL)full
 
 - (void)startFullReplay
 {
-    _sessionStart = _lastScreenShot;
-    _isFullSession = YES;
+    @synchronized(self) {
```

Comment:
`m`: Why do we need to synchronize here? What problem does it solve? Does it make sense to add a test for this?

Reply:
To avoid a race condition where two errors are captured very close to each other.
Honestly, I don't  trust tests for race conditions; they require too much 'magic' that makes them so flaky

Reply:
synchronization doesn't seem needed here though. DOesn't take any argument, two concurrent calls won't do anything. Scope changing is already thread safe.
And reading of the fields are isnt' atomic anyway